lib/wor/authentication/controllers/base.rb

```diff
@@ -0,0 +1,65 @@
+module Wor
+  module Authentication
+    module Controllers
+      module Base
+        def authenticate_request!
+          decoded_token.validate!(current_entity, entity_custom_validation_value(current_entity))
+        end
+
+        def decoded_token
+          @decoded_token ||=
+            Wor::Authentication::Token::Decoded.new(authentication_token, token_key)
+        end
+
+        def current_entity
+          @current_entity ||= find_authenticable_entity(decoded_token)
+        end
+
+        alias_method :current_user, :current_entity
+
+        def authentication_token
+          if request.headers['Authorization'].blank?
+            raise Wor::Authentication::Exceptions::MissingAuthorizationHeader
+          end
+          request.headers.fetch('Authorization', '').split(' ').last
+        end
+
+        ##########################################################################################
+        #                   DEFAULT METHOD IMPLEMENTATIONS, USER SHOULD CHANGE                   #
+        ##########################################################################################
+        def entity_custom_validation_value(entity)
+          if entity.respond_to?(:entity_custom_validation)
+            current_key = entity.entity_custom_validation
+            return current_key if current_key
+            update_custom_validation_value(entity)
+          end
+        end
+
+        def token_key
+          raise Wor::Authentication::Exceptions::NoKeyProvided if Wor::Authentication.token_key.blank?
+          Wor::Authentication.token_key
+        end
+
+        def find_authenticable_entity(decoded_token)
+          User.find_by(id: decoded_token[:id])
```

Comment:
If the user id doesn't exist, this will return nil instead of an throwing an exception. I think it should throw an exception, it can be an ActiveRecord::NotFound or one related to Wor authentication (to return an unathorized response instead of an 500)

Reply:
It does, when validating the token with the entity we raise a NoEntityPresent exception

Reply:
What happens with the cases in which you want to check the logged user but it is not required for a user to be logged?